src/domain/event/queryUtils.ts

```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocation } from 'react-router';
+import { toast } from 'react-toastify';
+
+import {
+  EventListQueryVariables,
+  useCourseListQuery,
+  useEventListQuery,
+} from '../../generated/graphql';
+import useLocale from '../../hooks/useLocale';
+import {
+  DEFAULT_SEARCH_FILTERS,
+  EVENT_SORT_OPTIONS,
+  PAGE_SIZE,
+} from '../eventSearch/constants';
+import {
+  getEventSearchVariables,
+  getNextPage,
+  getSearchQuery,
+} from '../eventSearch/utils';
+import { SIMILAR_EVENTS_AMOUNT } from './constants';
+import { getEventFields, getEventIdFromUrl } from './EventUtils';
+import { EventFields } from './types';
+
+const useSimilarEventsQueryVariables = (event: EventFields) => {
+  const locale = useLocale();
+  const { search } = useLocation();
+  const { keywords } = getEventFields(event, locale);
+  const eventSearch = getSearchQuery({
+    ...DEFAULT_SEARCH_FILTERS,
+    keyword: keywords.map((keyword) => keyword.id),
+  });
+
+  // Filter by search query if exists, if not filter by event keywords
+  const searchParams = new URLSearchParams(search || eventSearch);
+  return React.useMemo(() => {
+    return getEventSearchVariables({
+      include: ['keywords', 'location'],
+      language: locale,
+      pageSize: PAGE_SIZE,
+      params: searchParams,
+      sortOrder: EVENT_SORT_OPTIONS.END_TIME,
+      superEventType: ['umbrella', 'none'],
+    });
+  }, [locale, searchParams]);
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useSimilarEventsQuery = (event: EventFields) => {
```

Comment:
Could this be combined with `useSimilarCoursesQuery` ? 
```suggestion
export const useSimilarEventsQuery = (event: EventFields, useQuery = useEventListQuery, listType = 'eventList') => { ... }
```

Reply:
I think I initially thought something like this, but my biggest concern was that handling the types would be hard.

Reply:
I can test if I can come up with some solution with the typings. Maybe I will learn something new :D